### ISSUE TITLE
[Feature Request] In the L1/L2 cache, the L2 PUT and DELETE operations should be async fire and forget 

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-identity-web#1526]
As per discussion with SharePoint cache owners, they have gotten a perf boost by executing L2 operations in 'fire and forget" mode. Particularly for Redis, PUT operations take a long time. The L1 cache will have been updated anyway. 

Should only be done if L1 is enabled.

**Comment by jmprieur** [comment]
How is this achieved, @bgavrilMS ?
By removing the await and adding a pragma disable?

**Comment by bgavrilMS** [comment]
No, you'd do

```csharp
Task.Run(async () => ...)
```

Have a look at this helper function that MSAL uses to perform refresh_in operation in the background: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/master/src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequestHelper.cs#L70

**Comment by jennyf19** [comment]
We considered doing this before, but we discussed with @henrik-me and went with a re-try on failure. We'd have to see if they have set the re-try callback and then do fire and forget.

**Comment by bgavrilMS** [comment]
I was thinking about that as well. Maybe the re-try can be part of the fire and forget logic

```csharp
Task.Run(async() => 
bool success = WriteToRedis();
if (!success) 
   WriteToRedis();
);
```

**Comment by jennyf19** [comment]
closing as duplicate of #1047

**Comment by jennyf19** [comment]
Included in the [1.22.0 release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.22.00
